storybookjs/storybook
# Addon-docs: Fix 'show source' for stories with dynamic title

Issue: #9582 

## What I did
Do not refer to story title in addsMap keys - they are not needed.
Instead use just sanitized story name.
Snapshots were updated and manually compared because cases were reordered due to change in id (title was removed from key) 

## How to test

- Is this testable with Jest or Chromatic screenshots?
Yes, tests are updated
- Does this need a new example in the kitchen sink apps?
Yes, one story was added to how dynamic title exapmle
- Does this need an update to the documentation?
No this is a bug

If your answer is yes to any of these, please make sure to include it in your PR.
